botocore/exceptions.py

```diff
@@ -342,17 +342,30 @@ class UnsupportedSignatureVersionError(BotoCoreError):
 class ClientError(Exception):
     MSG_TEMPLATE = (
         'An error occurred ({error_code}) when calling the {operation_name} '
-        'operation: {error_message}')
+        'operation{retry_info}: {error_message}')
 
     def __init__(self, error_response, operation_name):
+        retry_info = self._get_retry_info(error_response)
         msg = self.MSG_TEMPLATE.format(
             error_code=error_response['Error'].get('Code', 'Unknown'),
             error_message=error_response['Error'].get('Message', 'Unknown'),
-            operation_name=operation_name)
+            operation_name=operation_name,
+            retry_info=retry_info,
+        )
         super(ClientError, self).__init__(msg)
         self.response = error_response
         self.operation_name = operation_name
 
+    def _get_retry_info(self, response):
+        retry_info = ''
+        if 'ResponseMetadata' in response:
+            metadata = response['ResponseMetadata']
+            if metadata.get('MaxAttemptsReached', False):
+                if 'RetryAttempts' in metadata:
+                    retry_info = (' (max_retries=%s reached)' %
```

Comment:
Yeah I played around with a few error message formats initially.  The main thing I was noticing is that specifically for the CLI, if you inject the retry info in the middle, and it's fairly long as you've suggested, the actual service error message is all the way on the right hand side and/or wraps to the next line.  Is their something more succinct we can say instead that disambiguates this from a CLI param?